examples/flow.cpp

```diff
@@ -150,6 +154,20 @@ try
         std::cout << "**********************************************************************\n\n";
     }
 
+#ifdef _OPENMP
+    if (!getenv("OMP_NUM_THREADS")) {
+        //Default to max(4,#cores) threads,
+        //not number of cores (unless ENV(OMP_NUM_THREADS) is defined)
+        int num_cores = omp_get_num_procs();
+        int num_threads = std::min(4, num_cores);
```

Comment:
Ok, a bit of a clumsy comment there (in the source code):

if you have 2 cores, you should use at most 2 threads, if you have 89 cores, you should use at most 4 threads. Code is right, comment is wrong :)